src/contact/components/EditInfo.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+
+export default class EditInfo extends React.Component {
+	constructor(props){
+		super(props);
+	}
+	componentDidMount(){
+		if(this.props.info === null){
+			this.props.actions.displayInfo(true);
+		} else {
+			this.props.actions.displayInfo(true,this.props.info);
+		}
+	}
+
+	addEmailField(){
+		this.props.actions.addEmail();
+	}
+
+	addPhoneField(){
+		this.props.actions.addTel();
+	}
+	
+	onInputChange = (e,field) => {
+		this.props.actions.handleInputChange({fieldname:e.target.name,value:e.target.value});
+		this.props.actions.validate({'fieldname':field, 'value':""});
+	}
+
+	//other emails change 
+	onEmailChange = (e) => {
+		this.props.actions.handleEmailChange({fieldname:e.target.name,value:e.target.value});
+	}
+
+	//other ph no change
+	onPhoneNoChange = (e) => {
+		this.props.actions.handlePhoneNoChange({fieldname:e.target.name,value:e.target.value});
+	}
+
+	cancel(){
+		this.props.actions.displayInfo(false);
+		this.props.history.push('/');
+	}
+
+	//custom validations
+	validations(){
+		const {firstname,lastname,primaryemail,primaryphno,otheremails,otherphnos} = this.props.info;
+		let flag = false;
+		if(firstname == '' || firstname.trim()==''){
+			this.props.actions.validate({'fieldname':'errorfn', 'value':"First name is required"});
+			flag=true;
+		}
+		if(lastname == '' || lastname.trim()==''){
+			this.props.actions.validate({'fieldname':'errorln', 'value':"Last name is required"});
+			flag=true;
+		}
+		if(primaryemail == '' || primaryemail.trim()==''){
+			this.props.actions.validate({'fieldname':'erroremail', 'value':"Primary email is required"});
```

Comment:
Can you make sure that you have a readable code by breaking different key value pairs in to different lines.